### PR TITLE
Console app with most of all refit calls with Postman Echo and a single usecase GraphQL call

### PR DESCRIPTION
**Pull request by Legendami**
We only have to implement test-GraphQL calls still. But so far we only have a simple get from the implementation itself, we should look into similar API's like Postman Echo for GraphQL that support more different types of calls for testing.

### CODE REVIEW

File: src/Xablu.WebApiClient.sln [hunk 1]
```diff
@@ -130,6 +131,7 @@ Global
 		{69AFB91C-FF88-45AD-BCCF-9D1F3B428111}.Release|x64.Build.0 = Release|Any CPU
 		{69AFB91C-FF88-45AD-BCCF-9D1F3B428111}.Release|x86.ActiveCfg = Release|Any CPU
 		{69AFB91C-FF88-45AD-BCCF-9D1F3B428111}.Release|x86.Build.0 = Release|Any CPU
+<<<<<<< HEAD
```
Comment (nmilcoff) [inline src/Xablu.WebApiClient.sln]: This is an unresolved merge conflict, can you please check this?

File: src/TestConsoleApp/TestConsoleApp.csproj [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\TestApp\TestApp\TestApp.csproj" />
+    <ProjectReference Include="..\Xablu.WebApiClient\Xablu.WebApiClient.csproj" />
+  </ItemGroup>
+  <ItemGroup>
+    <Reference Include="Mono.Android">
```
Comment (nmilcoff) [inline src/TestConsoleApp/TestConsoleApp.csproj]: How comes that a project targeting .NET Core has a reference to Mono.Android? 🤔 was it added by accident? Can we get rid of it?

File: src/TestConsoleApp/Program.cs [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                    break;
+            }
+        }
+
+        static async Task PrintRefitCall()
+        {
+            switch (Console.ReadLine())
+            {
+                case "1":
+                    Console.WriteLine("Wait for your result...");
```
Comment (nmilcoff) [inline src/TestConsoleApp/Program.cs]: Quick one: This statement is repeated in all switch cases :)

File: src/TestConsoleApp/RefitExampleCalls.cs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                    (ex) => true,
+                    3);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("This call is longer then 3 seconds, but will display anyway of working correctly");
+            }
+
+            ConvertJson(jsonresult);
+            return await Task.FromResult(result.Results);
```
Comment (nmilcoff) [inline src/TestConsoleApp/RefitExampleCalls.cs]: Here you can just say `return result.Results`

File: src/TestConsoleApp/RefitExampleCalls.cs [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+{
+    public class RefitExampleCalls
+    {
+        static List<Starships> items = new List<Starships>();
+
+        // 2 ways to check an object. One from json printed, the other an object from the api
+        // the first way is more modular and works for any json, the second more adjusted for SWAPI
+        public static async Task<IEnumerable<Starships>> GetStarShipItemsAsync(bool forceRefresh = false)
+        {
+            IWebApiClient<IStarwarsApi> _webApiClient = WebApiClientFactory.Get<IStarwarsApi>("https://swapi.co/api", false, () => new SampleHttpClientHandler());
```
Comment (nmilcoff) [inline src/TestConsoleApp/RefitExampleCalls.cs]: Not particularly wrong or anything like that, but is there any reason why you are passing along the `SampleHttpClientHandler` here?

File: src/TestConsoleApp/RefitExampleCalls.cs [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+            //    Xablu.WebApiClient.Enums.Priority.UserInitiated,
+            //    2,
+            //    (ex) => true,
+            //    60);
+
+            Console.WriteLine("\t basic auth: ");
+            ConvertJson(basicAuthResult);
+            //Console.WriteLine("\t oauth: ");
+            //ConvertJson(oAuthResult);
+            return await Task.FromResult(basicAuthResult);
```
Comment (nmilcoff) [inline src/TestConsoleApp/RefitExampleCalls.cs]: This can be `return basicAuthResult`

File: src/Xablu.WebApiClient/Services/Rest/RefitService.cs [hunk 7]
```diff
@@ -77,7 +77,7 @@ protected virtual void DisableAutoRedirects(HttpMessageHandler messageHandler)
                 {
                     Logger.Trace("Disabling AutoRedirects");
                 }
-                internalClientHandler.AllowAutoRedirect = false;
+                internalClientHandler.AllowAutoRedirect = true;
```
Comment (nmilcoff) [inline src/Xablu.WebApiClient/Services/Rest/RefitService.cs]: Is this change intended? I didn't see anything about it in the PR description, also the method where this line is located is called `DisableAutoRedirects` 🤔

File: src/Xablu.WebApiClient.sln (hunk 1 above)
Comment (Legendami) [reply src/Xablu.WebApiClient.sln]: This is baffling me, i had no more conflicts when pushing 🤔.

File: src/TestConsoleApp/RefitExampleCalls.cs (hunk 5 above)
Comment (Legendami) [reply src/TestConsoleApp/RefitExampleCalls.cs]: This is the way the calls are being used with your implementation in the testapp. I used those as a guideline!

Comment (Legendami) [reply src/TestConsoleApp/RefitExampleCalls.cs]: I tried implementing a new DelegatingHandler but that didn't work. Also i have no clue what this is even doing...

Comment (nmilcoff) [reply src/TestConsoleApp/RefitExampleCalls.cs]: Well, in that case I would suggest you search around for what `DelegatingHandler` is and what does it do... I can recommend you this [blog post](https://www.stevejgordon.co.uk/httpclientfactory-aspnetcore-outgoing-request-middleware-pipeline-delegatinghandlers) (just ignore the section "Registering Handlers" because it's for web dev).

Also debugging and setting breakpoints can help understand how does the flow go - and parameter names. The second parameter is what you want to have set to `true` instead of changing the bool value in the `DisableAutoRedirects` method, and the third parameter in this case is unnecessary

**Comment by Legendami** [comment]
Now with a final GraphQL call and implemented requested changes

**Comment by Legendami** [comment]
I have no clue what goes wrong with the build here, i properly rebased this branch on develop
